netlink-packet/src/sock_diag/buffer.rs

```diff
@@ -0,0 +1,85 @@
+use std::mem;
+
+use byteorder::{ByteOrder, NativeEndian};
+
+use crate::Field;
+
+pub const SDIAG_FAMILY: usize = 0;
+pub const SDIAG_PROTOCOL: usize = 1;
+
+pub const fn array_of<T>(start: usize, len: usize) -> Field {
+    start..(start + mem::size_of::<T>() * len)
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct RtaIterator<T> {
+    position: usize,
+    buffer: T,
+}
+
+impl<T> RtaIterator<T> {
+    pub fn new(buffer: T) -> Self {
+        RtaIterator {
+            position: 0,
+            buffer,
+        }
+    }
+}
+
+const RTA_ALIGNTO: usize = 4;
+const RTA_HDR_LEN: usize = mem::size_of::<u16>() * 2;
+
+const RTA_LENGTH: Field = 0..2;
+const RTA_TYPE: Field = 2..4;
+
+impl<'buffer, T: AsRef<[u8]> + ?Sized + 'buffer> Iterator for RtaIterator<&'buffer T> {
```

Comment:
This looks similar (but not exactly the same) to [`NlaIterator`](https://github.com/little-dude/netlink/blob/fd5e8d37adb75c1dc4bc3c20bcbc576a5389035a/netlink-packet/src/rtnl/nla.rs#L228). Do you think the whole `nla.rs` would be worth re-using instead?

Reply:
yes, I think we should extract those reusable code for general netlink protocol, but maybe later